Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -664,16 +669,221 @@ namespace xr
+            ArFrame_getUpdatedTrackables(xrContext->Session, xrContext->Frame, AR_TRACKABLE_AUGMENTED_IMAGE, trackableList);
+            ArTrackableList_getSize(xrContext->Session, trackableList, &imageListSize);
+
+            // For each updated image, get the current status.
+            for (int i = 0; i < imageListSize; ++i) {
+                ArTrackable* trackable = nullptr;
+                ArTrackableList_acquireItem(xrContext->Session, trackableList, i, &trackable);
+                ArAugmentedImage* imageTrackable = ArAsAugmentedImage(trackable);
+
+                int imageIndex;
+                ArAugmentedImage_getIndex(xrContext->Session, imageTrackable, &imageIndex);
+
+                float measuredWidthInMeters;
+                ArAugmentedImage_getExtentX(xrContext->Session, imageTrackable, &measuredWidthInMeters);
+                
+                float rawPose[7];
+                ArAugmentedImage_getCenterPose(xrContext->Session, imageTrackable, tempPose);
+                ArPose_getPoseRaw(xrContext->Session, tempPose, rawPose);
+
+                ArAugmentedImageTrackingMethod trackingState{AR_AUGMENTED_IMAGE_TRACKING_METHOD_NOT_TRACKING};
+                ArAugmentedImage_getTrackingMethod(xrContext->Session, imageTrackable, &trackingState);
+
+                // Update the existing image tracking result if it exists.
+                auto resultIterator{ imageTrackingResultsMap.find(imageTrackable) };
+                if (resultIterator != imageTrackingResultsMap.end())
+                {
+                    UpdateImageTrackingResult(
+                        updatedResults,
+                        GetImageTrackingResultByID(resultIterator->second),
+                        rawPose,
+                        measuredWidthInMeters,
+                        trackingState);
+
+                    // Release reference to trackable, since we are already holding a ref count in the map.
+                    ArTrackable_release(reinterpret_cast<ArTrackable*>(imageTrackable));
+                }
+                else
+                {
+                    // This is a new result, create it and initialize its values.
+                    imageTrackingResults.push_back(std::make_unique<Frame::ImageTrackingResult>());
+                    auto& result{ *imageTrackingResults.back() };
+                    result.Index = imageIndex;
+                    imageTrackingResultsMap.insert({imageTrackable, result.ID});
+                    UpdateImageTrackingResult(
+                        updatedResults,
+                        result,
+                        rawPose,
+                        measuredWidthInMeters,
+                        trackingState);
+                }
+            }
+        }
+
+        Frame::ImageTrackingResult& GetImageTrackingResultByID(Frame::ImageTrackingResult::Identifier resultID)
+        {
+            for (std::unique_ptr<Frame::ImageTrackingResult>& resultPtr : imageTrackingResults)
```

Comment:
Maybe use `std::find_if`. We use it elsewhere so there should be examples.

Reply:
Did not know this existed, will also use this for planes!